### DISCUSSION TITLE
PIC16 .sinc files

### DISCUSSION THREAD

**Discussion by RobertGreenlight** [NationalSecurityAgency/ghidra#4432] [Q&A]
Hello, first time here, and new to Ghidra.  I am trying to disassemble a PIC16C72A and the commands for TMR1H and TMR1L don't work. The disassembler thinks that the hex locations of 0E and 0F are values, not addresses.  I looked a the .sinc file and these are missing.  Is there a way to manually enter these commands or is there a way I can add them to the .sinc file myself?

**Comment by ghidra1** [comment]
There are so many PIC variants we do not define all the variants and their specific registers.  If loading an ELF file these may be defined as symbols.  If so, changes were made to symbol address calculations for `SHN_ABS` symbols which load these symbols properly.  See 779635aec2c6446ab8cd8b9ee62481265c486d18.   Otherwise, you can add the memory labels yourself; either directly as labels or to the appropriate `*.pspec` file or add your own variant definition (see `pic16.ldefs`).   You should not need to modify the *.sinc file which is intended to remain generic.
